### DISCUSSION TITLE
getServerSideProps not being called on imported Nextjs Pages

### DISCUSSION THREAD

**Discussion by 1finedev** [vercel/next.js#25093] [Help] [answered]
### What version of Next.js are you using?

v10.0.9

### What version of Node.js are you using?

v14.15.4

### What browser are you using?

Chrome

### What operating system are you using?

macOs

### How are you deploying your application?

vercel

### Describe the Bug

getServerSideProps is not being called on the imported Component in Nextjs i need help!

it only works when i visit the page url directly but if imported it doesnt call getServerSideProps on the imported page

### Expected Behavior

Expect getServerSideProps to run on every request

### To Reproduce

Here is my Coupon Page

```
const Coupon = () => {
  return (
    <div>
      <p>Hello</p>  
    </div>
  );
};

// coupon page
export async function getServerSideProps(context) {
  console.log(context);
  console.log("here");

  return {
    props: {}, // will be passed to the page component as props
  };
}
export default Coupon;
```

and my Layout Page

```
import Coupon from './coupon'

const Layout = () => {
  return (
    <div>
      <Coupon />
    </div>
  );
};

export default Layout;
```

**Comment by dlaugharnTAG** [comment, marked as answer]
This isn't a bug, getServerSideProps needs to be an explicitly named export on the component in pages/

So if, say Layout and Coupon lived in a components directory, a pages/index.js component would look like this:

```js
import { getServerSideProps } from '../components/Coupon'
import Layout from '../components/Layout'

export default Layout

export {
  getServerSideProps
}
```

**Comment by 1finedev** [comment]
Thanks

**Comment by 1finedev** [comment]
But the props passed into the coupon page does not work when imported still

**Comment by dlaugharnTAG** [comment]
The props belong to the top level page component. If pages/index.js is exporting Layout, Layout has the props, and those will need to be passed down into Coupon.

**Comment by 1finedev** [comment]
i mean i used the data from getServerSideProps in the coupon page and whenever i load the Layout page which imports the coupon page, the data is never fetched from the server hence coupon page gets undefined props from getServerSideProps

**Comment by 1finedev** [comment]
i really pray you understand my issue this time

i have two pages 

layout which imports Coupon 
coupon uses props from data gotten from getServerSideProps   

whenever i load Layout 
Coupon renders without the props from its getServerSideProps  function
